tests/testthat/test-set_view_lim.R

```diff
@@ -0,0 +1,41 @@
+context("set_view_lim")
+
+test_that("gsplot xaxs = 'o' works", {
+  gs <- gsplot() %>%
+    points(1:4,1:4, xlim=c(NA,10), xaxs='o') %>%
+    lines(1:6,1:6)
+  
+  beforeBuffer <- c(1,10)
+  buffer <- diff(beforeBuffer)*0.04
+  afterBuffer <- c(beforeBuffer[1]-buffer, beforeBuffer[2])
+  
+  expect_equal(xlim(gs), afterBuffer)
+  expect_equal(par(gs)$xaxs, "i")
+})
+
+
+test_that("gsplot yaxs = 'o' works", {
+  gs <- gsplot() %>%
+    points(1:4,1:4, ylim=c(0,NA), yaxs='o') %>%
+    lines(1:6,1:6)
+  
+  beforeBuffer <- c(0,4)
+  buffer <- diff(beforeBuffer)*0.04
+  afterBuffer <- c(beforeBuffer[1], beforeBuffer[2]+buffer)
+  
+  expect_equal(ylim(gs), afterBuffer)
+  expect_equal(par(gs)$yaxs, "i")
+})
+
+test_that("gsplot yaxs = 'o' works", {
+  gs <- gsplot(ylim=c(0,NA), yaxs='o') %>%
+    points(1:4,1:4) %>%
+    lines(1:6,1:6)
+  
+  beforeBuffer <- c(0,4)
+  buffer <- diff(beforeBuffer)*0.04
+  afterBuffer <- c(beforeBuffer[1], beforeBuffer[2]+buffer)
+  
+  expect_equal(ylim(gs), afterBuffer)
+  expect_equal(par(gs)$yaxs, "i")
+})
```

Comment:
@ldecicco-USGS @jread-usgs 

Current workflow for yaxs='o' is to include it in the call for points/lines etc. It will buffer the limit of your data in that call, but if you add data after that axis call, it will not be included in the buffer. Should we change the workflow so that it goes in `gsplot()` and therefore applies to all data (see third test for example)?